tests/browser/test_tools.py

```
@@ -52,16 +68,78 @@ def test_tool_execution_in_session(persona: NottePersona, action: EmailReadActio
+                        f"sender={email.sender_email!r}",
+                        f"created_at={email.created_at.isoformat()}",
+                    )
+
+                matching_emails = [
+                    email for email in emails if email.created_at >= started_at and email.subject == subject
+                ]
+                if matching_emails:
+                    return
+
+            subjects = [email.subject for email in emails]
+            assert False, f"No fresh SMTP test email found in {len(emails)} emails. Subjects: {subjects!r}"
```

Ruff (0.15.12)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)